### ISSUE TITLE
Code reference search: unclear how to apply filters

### ISSUE THREAD

**Issue by ryelle** [WordPress/wporg-developer#200]
When on the search results page, if I want to apply a filter (for example, only show functions), I can click the checkbox, but it's not clear how to apply that filter. I ended up clicking back into the search box and hitting enter.

The little magnifying glass is technically the search button, so it submits the filters too, but it doesn't look like a button, and since it's contained in the input it looks like it's only for the text field.

<img width="745" alt="Screenshot 2023-02-03 at 1 38 32 PM" src="https://user-images.githubusercontent.com/541093/216681501-a4a05007-6ddb-4067-9f10-fa57c17f6b33.png">

The checkbox should not automatically submit the form, as that causes a page load and would be bad for a11y/UX.

cc @WordPress/meta-design

**Comment by StevenDufresne** [comment]
The taxonomy archive has the same problem:

![image](https://user-images.githubusercontent.com/1657336/216885773-66b745a8-268f-41c1-b21c-fb2e4ffc2d44.png)

[Figma](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?node-id=1160%3A10164&t=oU0nxF52BZkGOBc3-0)

**Comment by StevenDufresne** [comment]
Here's a not-so-nice take on it, a button!

<img width="749" alt="Screen Shot 2023-02-06 at 2 11 09 PM" src="https://user-images.githubusercontent.com/1657336/216888749-4161409b-07f3-4a3b-a0bc-822de7a64f5f.png">

Branch: [fix/button-to-search](https://github.com/WordPress/wporg-developer/tree/fix/button-to-search)

**Comment by jasmussen** [comment]
This issue exists in production already, correct? 
<img width="1028" alt="Screenshot 2023-02-06 at 09 20 56" src="https://user-images.githubusercontent.com/1204802/216920406-a8b03bf7-04fc-4813-95ea-1235eeb6dd57.png">

Since the search icon is technically a button, is it a bit awkward, tabbing-wise, to have two submit buttons that do the same, but have different labels? 

Openverse reloads the page immediately on click of a filter, though that appears to be using some React smarts so focus isn't lost. But that seems like a better longer term concept. 

We can make the apply button work if need be, but it strikes me as extra UI so it'd be nice to look at best practices in this space before we go that route. What do other non-react based search engines do?

**Comment by ryelle** [comment]
> This issue exists in production already, correct?

So it does. We should still fix it 🙂 

> Since the search icon is technically a button, is it a bit awkward, tabbing-wise, to have two submit buttons that do the same, but have different labels?

True - in this case, I'd prefer removing the button from the search input in favor of a more visible one that clearly includes the filter selections.

> Openverse reloads the page immediately on click of a filter, though that appears to be using some React smarts so focus isn't lost. But that seems like a better longer term concept.

As far as I know, we have no intention to implement this as a react-controlled page. There are also plenty of other a11y issues that pop up when you go that route.

> What do other non-react based search engines do?

We could look at [learn](https://learn.wordpress.org/lesson-plans/?search=test), which uses an apply button in the sidebar, but has many more filters. There's also [github](https://github.com/search?q=test), which also has a sidebar but they use links instead so there's no second button. Maybe a tabs-style approach could work? if they're links it would be expected that clicking will trigger an action (page reload).

<img width="948" alt="Screenshot 2023-02-06 at 1 24 24 PM" src="https://user-images.githubusercontent.com/541093/217054209-a8d5ff90-cd85-4c9f-951d-70894e710630.png">

I've looked around for other examples but they're either much more complex (like learn), and/or they do the JS filtering approach.

**Comment by ryelle** [comment]
I've just realized that the tab idea won't help on the taxonomy page — that one really does just need a button 🙂

**Comment by jasmussen** [comment]
Another option is to have the search field expand into a larger box, like browsers do with the URL bar, and have filters exist inside that. But this needs a larger design, and a separate iteration, so it's just something to consider. For now, and since this is a subpage, let's just go with Steve's suggestion, but use a blue button instead:

<img width="770" alt="Screenshot 2023-02-07 at 08 58 21" src="https://user-images.githubusercontent.com/1204802/217185771-17ed60fa-c357-4b82-8269-f7ac03683847.png">

**Comment by adamwoodnz** [comment]
Should be tested as part of https://github.com/WordPress/wporg-developer/pull/297

**Comment by adamwoodnz** [comment]
This was fixed in #292, where the 'Apply' button was added to the filters block

![Screenshot 2023-10-06 at 10 40 07 AM](https://github.com/WordPress/wporg-developer/assets/1017872/62c9f66e-c15a-4f30-9fd3-e24a80ac89f7)
